tensorrt_llm/_torch/auto_deploy/transformations/library/fusion.py

```diff
@@ -110,6 +136,31 @@ def split_output(tensor: torch.Tensor) -> Tuple[torch.Tensor, ...]:
     gm.delete_all_unused_submodules()
```

Comment:
Can you delete the old unused weight parameter directly?

Otherwise, we get at `O(n^2` since these utilities also iterate over all nodes, where `n = # nodes`

(same as other comment)